RHIDP-1037: Creating RN for RHDH 1.1 (#79)
redhat-developer/red-hat-developers-documentation-rhdh: release-notes/con-relnotes-notable-features.adoc

=== Configuration change in the GitHub Org catalog provider  

The GitHub Org catalog module for the new backend system switched from an obsolete implementation to a new one, that is, `GithubMultiOrgProvider`. This new implementation brings in changes to the default plugin configuration settings.

=== New Scaffolder Identity client

Identity client functioning differs slightly between the one provided by the new backend system Scaffolder plugin, and the one used in the previous release:

* The previous release uses the `DefaultIdentityClient` function without any specific option
* New backend system Scaffolder does not use the default identity service, but still uses the deprecated scaffolder-dedicated `buildDefaultIdentityClient` function, which introduces a slight functioning change in the identity client. 

=== Enable all GitLab actions
{product} {product-version} enables all GitLab actions by default.

> === Events Plugin support broken in the new backend system
> Events usefulness is greatly reduced, if not completely broken, on the new backend system because cross-plugin modules are not supported. This limitation prevents third parties from fully using events in plugins or modules they want to provide as dynamic plugins for {product}.

Reviewer: ```suggestion
=== Events support redesign in progress for the new backend system
The Events plugin is not completely supported by the new backend system, and thus its usefulness is greatly reduced, if not completely broken.
A complete redesign of the Events support for the new backend system is in progress, but was still not available to integrate it to this release of {product}. This limitation temporarily prevents third parties from fully using events in plugins or modules they want to provide as dynamic plugins for {product}.
```

This section highlights new features in {product} {product-version}.

> == Support for upgrading the {product} Helm Chart 

Reviewer 2: This is not a notable new feature. The upgrade was supported from the beginning; there was just nothing to upgrade from, as we had only one release.

Author: @kadel, I understand your point. Since we did not have the upgrade procedure mentioned in the admin guide, we were planning to cover it in the Release Notes. Would it be okay if i move this piece of content in the previous chapter [about this release](https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.0/html-single/release_notes_for_red_hat_developer_hub_1.0/index#con-release-notes-overview_release-notes-rhdh).

> 

Reviewer 2: Instead of the Helm Chart information I would mention that RHDH is now based on the Backstage v1.23.4. that that is more important information

Author: == Change in productized version  

{product} is now a fully supported, enterprise-grade productized version of upstream Backstage v1.23.4.

+
[source, CLI]
----
$ helm upgrade -i rhdh -f new-values.yml \
  openshift-helm-charts/redhat-developer-hub --version 1.1.0
----
+
[NOTE]
====
You can also provide extra values to the chart by creating a `new-values.yml` file on your workstation with values that override the attributes in the installed chart, or add new attributes.
====

== Ability to manage role-based access controls (RBAC) using the web interface  

As an administrator, you can now use {product-short} web interface (Web UI) to assign specific roles and permissions to individual users or groups. Using the {product-short} web UI, you can perform the following actions:

* Creating a role 
* Editing a role and related permissions
* Deleting a role

==  Migration of {product} to the new backend system  

{product} {product-version} is now migrated to the Backstage new backend system. This migration in combination with the dynamic plugins feature allows you to extend your backend plugins with an unprecedented ease. With this migration, you might notice the following functionality-related changes in the {product-short} application.

> === Configuration change in the GitHub Org catalog provider  

Reviewer: In think it is. The migration to te new backend system involved doing this switch.
It's a the following sentence after:
> you might notice the following functionality-related changes in the {product-short} application

> === Configuration change in the GitHub Org catalog provider  

Author: Keeping the heading at the same level

=== Enable all GitLab actions
{product} {product-version} enables all GitLab actions by default.

=== Events support redesign in progress for the new backend system
The `Events` plugin is not completely supported by the new backend system, and thus its usefulness is greatly reduced, if not completely broken.
A complete redesign of the `Events` plugin support for the new backend system is in progress, but was still not available to integrate it to this release of {product}. This limitation temporarily prevents third parties from fully using events in plugins or modules they want to provide as dynamic plugins for {product}.

=== RBAC Plugin
In the new backend system, automatically getting the IDs of all dynamically installed plugins is not possible. This means that permissions of dynamic plugins supporting RBAC will not be automatically displayed in the RBAC UI.
As a workaround, you can manually add a list of dynamic plugin IDs supporting RBAC in the {product-short} application configuration file.

== Support for Elastic Kubernetes Services (EKS)

> You can now integrate your {product} application with Amazon Web Services (AWS), which optimizes your development workflows within the AWS ecosystem.

Reviewer 2: Similarly as with EKS, this is not really true. The EKS support is mainly about supporting installing RHDH into EKS clusters. There is no optimizations of the development workflow (not even sure it would mean :-/ )

Author: For better clarity, rephrasing this part to:
You can now install and use {product} on an EKS cluster.

> You can now integrate your {product} application with Amazon Web Services (AWS), which optimizes your development workflows within the AWS ecosystem.

Reviewer 3: ```suggestion
You can now integrate your {product} application with Amazon Web Services (AWS), which enables your development workflows within the AWS ecosystem.
```

Author: I have removed this line based on Tomas's suggestion as the statement does not depict the correct informantion.

> To enable this integration, deploy {product-short} in Elastic Kubernetes Services (EKS) using Helm Chart or Operator. 

Reviewer 2: Lets remove this line completely

Author: Removed line no 86 from the content

Author: For better clarity, rephrasing this part to:
You can now install and use {product} on an EKS cluster.

> To enable this integration, deploy {product-short} in Elastic Kubernetes Services (EKS) using Helm Chart or Operator. 

Reviewer 3: A blank line suggestion achieves that:

```suggestion
```

Author: Removed this line.

== Support for Azure Kubernetes Services (AKS)

> You can now seamlessly integrate Azure Kubernetes Services into your development workflow in {product}. Integrating {product-short} with AKS can simplify the deployment, management, and scaling of the application on Microsoft Azure’s fully managed Kubernetes service. 

Author: For better clarity, rephrased the sentence:
You can now install and use {product} on an AKS cluster.

== Support for viewing installed plugins using Web UI

You can now use {product-short} web interface (Web UI) to view a table of plugins that are installed. This feature uses the `dynamic-plugins-info` frontend component, which offers a table of plugins that are currently installed in {product} or Janus IDP. You can apply client-side sorting, filtering, and pagination on the plugins table.

== Supported plugins in {product} {product-version}

The following table lists the supported plugins in {product} {product-version}:

[id="supported-plugins-in-rhdh"]
.Supported plugins in {product} {product-version}
[cols="30%,30%,10%,30%",options="header",subs="+attributes"]
|===

> | Plugin name | Plugin ID | Version | Category

Reviewer 2: "Category" column probably doesn't make sense if it is only in one plugin

Reviewer 3: rather than having the list of plugins in a **THIRD** place, maybe we could just link to the main table in ref-rhdh-plugins.adoc ? https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/75/files#diff-48d14956b347986734708cd4e5df54eadbc40ea3c686e2b1b2e0088fb3f2a655

Author: I agree with Nick's suggestion. better to provide a link to that table here. Doing the same.

This section highlights new features in {product} {product-version}.

== Change in productized version  

{product} productized version is now based on the upstream Backstage project v1.23.4. 

== Ability to manage role-based access controls (RBAC) using the web interface  

As an administrator, you can now use {product-short} to assign specific roles and permissions to individual users or groups. Using the {product-short} web interface, you can perform the following actions:

* Creating a role 
* Editing a role and related permissions
* Deleting a role

==  Migration of the {product} to the new backend system  

{product} {product-version} is now migrated to the Backstage new backend system. With this migration, you might notice the following functionality-related changes in the {product-short} application:

=== Configuration change in the GitHub Organization catalog provider  

> The GitHub Organization catalog module for the new backend system switched to a new `GithubMultiOrgEntityProvider` provider, which allows ingesting data from multiple GitHub organizations. This new provider brings in changes to the default plugin configuration settings.

Author: @Gerry-Forde @hmanwani-rh, Have we covered anything related to this change in our documentation?

> The GitHub Organization catalog module for the new backend system switched to a new `GithubMultiOrgEntityProvider` provider, which allows ingesting data from multiple GitHub organizations. This new provider brings in changes to the default plugin configuration settings.

Reviewer 4: @kim-tsao, these behavior changes weren't covered in the documentation as they were raised recently. Therefore, we decided to include them in the release notes for RHDH 1.1.

> The GitHub Organization catalog module for the new backend system switched to a new `GithubMultiOrgEntityProvider` provider, which allows ingesting data from multiple GitHub organizations. This new provider brings in changes to the default plugin configuration settings.

Reviewer: The change in the default configuration of the embedded plugin is this one: https://github.com/janus-idp/backstage-showcase/commit/1f5f08073dba49289c418b746166ec14cdd30402#diff-2a8ec8751c51963220c18905adc57407d70bbb6647e6fe394cc224291b93a11fR15-R28

After looking deeper in the existing documentation, it seems that the following places should be updated:
- In the **_Dynamic plugins support matrix_** table in https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.0/html/administration_guide_for_red_hat_developer_hub/rhdh-installing-dynamic-plugins#rhdh-supported-plugins: The "Required Variables" column for the `@backstage/plugin-catalog-backend-module-github-org` plugin should be changed from: 1 variable (`GITHUB_ORG_URL`) to 2 variables (`GITHUB_URL`, and `GITHUB_ORG`).
- In the [**Github organization member discovery**](https://access.redhat.com/documentation/en-us/red_hat_developer_hub/1.0/html/getting_started_with_red_hat_developer_hub/ref-rhdh-supported-configs_rhdh-getting-started#enabling-github-organization-member-discovery-in-red-hat-developer-hub) section of the Getting Started docs, change the following:
>```
>kind: ConfigMap
>apiVersion: v1
>metadata:
>  name: app-config-rhdh
>data:
>  app-config-rhdh.yaml: |
>    ...
>    catalog:
>      providers:
>        githubOrg:
>          default:
>            id: production
>            orgUrl: '${GITHUB_ORG_URL}'
>    ...
>``` 
>In the previous code, replace ${GITHUB_ORG_URL} with the GitHub organization you want to ingest users from.

to :

>```
>            id: production
>            githubUrl: "https://github.com"
>            orgs: [ "${GITHUB_ORG}" ]
>```
>In the previous code, replace ${GITHUB_ORG} with the GitHub organization you want to ingest users from.

> The GitHub Organization catalog module for the new backend system switched to a new `GithubMultiOrgEntityProvider` provider, which allows ingesting data from multiple GitHub organizations. This new provider brings in changes to the default plugin configuration settings.

Author: @kim-tsao, Based on David's comment, i have updated the existing doc. It is basically about change in the variable name in the plugin config setting.

=== New Scaffolder Identity client

Identity client functioning differs slightly between the one provided by the new backend system Scaffolder plugin and the one used in the previous release:

* The previous release used the `DefaultIdentityClient` function without any specific option

> * The new backend system Scaffolder does not use the default identity service, but still uses the deprecated scaffolder-dedicated `buildDefaultIdentityClient` function, which introduces a slight functioning change in the identity client. 

Reviewer: It seems to me that your proposal sounds correct. As for for it means in more details for the enduser experience, maybe @kadel can answer better that me since he tested it.

=== Enable all GitLab actions

{product} {product-version} enables all GitLab actions by default.

== Support for Elastic Kubernetes Services (EKS)

You can now install and use the {product} on an EKS cluster. 

== Support for Azure Kubernetes Services (AKS)

You can now install and use the {product} on an AKS cluster. 

== Support for viewing installed plugins using the web interface 

> You can now use the {product-short} web interface to view a table of plugins that are installed. This feature uses the `dynamic-plugins-info` frontend component, which generates a table of plugins that are currently installed in the {product}. You can apply client-side sorting, filtering, and pagination to the plugins table.

Reviewer 5: This is an admin only view, we should call that out

This section highlights new features in {product} {product-version}.

== Change in productized version  

> {product} is now a fully supported, enterprise-grade productized version of upstream Backstage v1.23.4. 

Reviewer 2: This sounds like it was not supported before, which is not true.

Author: {product} productized version is now based on the upstream Backstage project v1.23.4.

== Ability to manage role-based access controls (RBAC) using the web interface  

As an administrator, you can now use {product-short} to assign specific roles and permissions to individual users or groups. Using the {product-short} web interface, you can perform the following actions:

* Creating a role 
* Editing a role and related permissions
* Deleting a role

==  Migration of {product} to the new backend system  

> {product} {product-version} is now migrated to the Backstage new backend system. This migration in combination with the dynamic plugins feature allows you to extend your backend plugins with an unprecedented ease. With this migration, you might notice the following functionality-related changes in the {product-short} application:

Reviewer 6: Is it correct to refer to "the **Backstage** new backend system"? 

@davidfestal can you confirm please?

Reviewer 6: Yes

Reviewer 2: I don't think that mentioning the underlying upstream project is a problem. If that what you are asking. 

Maybe it is just me, but the "_This migration in combination with the dynamic plugins feature allows you to extend your backend plugins with an unprecedented ease_" sentence sounds weird to me.
I've noticed this style in other places in release notes as well. It is sounds like marketing BS mixed with sales pitch :-/ It is something that I wouldn't expect in release notes. But it could be just me. I'm really alergic to empty marketing terms like "an unprecedented ease"

Reviewer 6: Yes, I'd agree that the customer may have a different view of the "unprecedented ease" statement. It's probably better to remove this phrase or the sentence altogether.

Author: Removed this sentence

=== Configuration change in the GitHub Org catalog provider  

> The GitHub Org catalog module for the new backend system switched from an obsolete implementation to a new one, that is, `GithubMultiOrgProvider`. This new implementation brings in changes to the default plugin configuration settings.

Author: i think it is part of this feature. For ref: https://backstage.io/docs/integrations/github/org/
I think this word is needed, but i have used Organization for better clarity.

> The GitHub Org catalog module for the new backend system switched from an obsolete implementation to a new one, that is, `GithubMultiOrgProvider`. This new implementation brings in changes to the default plugin configuration settings.

Reviewer 2: Another point: I don't think that the `GithubOrgEntityProvider` that was used before is deprecated. It is just that this provider allows to configure only one gihtub org. 
The one that we switched to (GithubMultiOrgEntityProvider) allows ingesting data from multiple GitHub organizations (more at https://backstage.io/docs/integrations/github/org/)

> The GitHub Org catalog module for the new backend system switched from an obsolete implementation to a new one, that is, `GithubMultiOrgProvider`. This new implementation brings in changes to the default plugin configuration settings.

Author: The GitHub Organization catalog module for the new backend system switched to a new `GithubMultiOrgEntityProvider` provider, which allows ingesting data from multiple GitHub organizations. This new provider brings in changes to the default plugin configuration settings.

=== New Scaffolder Identity client

Identity client functioning differs slightly between the one provided by the new backend system Scaffolder plugin, and the one used in the previous release:

* The previous release uses the `DefaultIdentityClient` function without any specific option

> * New backend system Scaffolder does not use the default identity service, but still uses the deprecated scaffolder-dedicated `buildDefaultIdentityClient` function, which introduces a slight functioning change in the identity client. 

Reviewer 7: Hello @shipsing , 

a) Could we be consistent? The first bullet doesn't have a period, but the second bullet does.
b) I believe, we should put a comma before **_but_** only when **_but_** is connecting two independent clauses. Here, comma before but can be dropped it seems? 

Please let me know your thoughts. Thanks!

> * New backend system Scaffolder does not use the default identity service, but still uses the deprecated scaffolder-dedicated `buildDefaultIdentityClient` function, which introduces a slight functioning change in the identity client. 

Author: I have added a period for the first point. Yes, for the second point, we do not need a comma because they are not independent clauses. So, leaving as is.

=== Enable all GitLab actions

> {product} {product-version} enables all GitLab actions by default.

Reviewer 6: Is this specifically related to the new backend system?

Author: This JIRA: https://issues.redhat.com/browse/RHIDP-1326 points to the same.

> === Events support redesign in progress for the new backend system

Reviewer 2: This, to me, looks more like a known issue than a new feature, and the same for the next section about "RBAC Plugin". I see that it is here under the new backend section, but wouldn't make more sense to move this to a Known issues section? @davidfestal?

Author: @davidfestal, I think we should move both these points to the known issues section because they are kind of work-in-progress items, which we want our users to know about, rather presenting them as new features as part of the migration.

The `Events` plugin is not completely supported by the new backend system, and thus its usefulness is greatly reduced, if not completely broken.

> A complete redesign of the `Events` plugin support for the new backend system is in progress, but was still not available to integrate it to this release of {product}. This limitation temporarily prevents third parties from fully using events in plugins or modules they want to provide as dynamic plugins for {product}.

Reviewer: It's simply not available at all in the RHDH 1.1 release (though it has already been merged in the backstage release 1.24 - underlying upstream project).

> A complete redesign of the `Events` plugin support for the new backend system is in progress, but was still not available to integrate it to this release of {product}. This limitation temporarily prevents third parties from fully using events in plugins or modules they want to provide as dynamic plugins for {product}.

Author: This is just for customers' knowledge. Just to make them aware about this.

=== RBAC Plugin

> In the new backend system, automatically getting the IDs of all dynamically installed plugins is not possible. This means that permissions of dynamic plugins supporting RBAC will not be automatically displayed in the RBAC UI.

Reviewer 7: ```suggestion
In the new backend system, automatically getting the IDs of all dynamically installed plugins is not possible. This means that the permissions of dynamic plugins supporting RBAC will not be automatically displayed in the RBAC UI.
```

As a workaround, you can manually add a list of dynamic plugin IDs supporting RBAC in the {product-short} application configuration file.

== Support for Elastic Kubernetes Services (EKS)

> You can now install and use {product} on an EKS cluster. 

Reviewer 7: ```suggestion
You can now install and use the {product} on an EKS cluster. 
```

Author: We do not need an article with the product name, specifically if we have not used any entity type after its actual name.

== Support for Azure Kubernetes Services (AKS)

> You can now install and use {product} on an AKS cluster. 

Reviewer 7: ```suggestion
You can now install and use the {product} on an AKS cluster. 
```

Author: same as above

== Support for viewing installed plugins using the web interface 

> You can now use the {product-short} web interface to view a table of plugins that are installed. This feature uses the `dynamic-plugins-info` frontend component, which generates a table of plugins that are currently installed in {product}. You can apply client-side sorting, filtering, and pagination on the plugins table.

Reviewer 7: ```suggestion
You can now use the {product-short} web interface to view a table of plugins that are installed. This feature uses the `dynamic-plugins-info` frontend component, which generates a table of plugins that are currently installed in the {product}. You can apply client-side sorting, filtering, and pagination to the plugins table.
```

> == Supported plugins in {product} {product-version}

Reviewer 6: Are we not listing new plugins added in this release?

Reviewer 6: @shipsing I see your point in relation to this section of the RNs. 

A more general question is if we need to identify to customers which **new plugins** have been added in each release, either in the RNs or in the Admin Guide table of dynamic plugins.

@kadel @davidfestal Could you advise please?
